jetty-server/src/main/java/org/eclipse/jetty/server/handler/ResourceHandler.java

```diff
@@ -140,44 +142,51 @@ public MimeTypes getMimeTypes()
     }
 
     @Override
-    public Resource getResource(String path)
+    public Resource getResource(String path) throws IOException
     {
         if (LOG.isDebugEnabled())
-            LOG.debug("{} getResource({})", _context == null ? _baseResource : _context, _baseResource, path);
+            LOG.debug("{} getResource({}): baseResource:{}", _context == null ? _baseResource : _context, path, _baseResource);
 
-        if (path == null || !path.startsWith("/"))
-            return null;
+        if (StringUtil.isBlank(path))
+        {
+            throw new IllegalArgumentException("Path is blank");
+        }
 
-        try
+        if (!path.startsWith("/"))
+        {
+            throw new IllegalArgumentException("Path reference invalid: " + path);
+        }
+
+        Resource r = null;
+
+        if (_baseResource != null)
         {
-            Resource r = null;
+            path = URIUtil.canonicalPath(path);
+            r = _baseResource.addPath(path);
 
-            if (_baseResource != null)
+            if (r.isAlias() && (_context == null || !_context.checkAlias(path, r)))
             {
-                path = URIUtil.canonicalPath(path);
-                r = _baseResource.addPath(path);
-
-                if (r != null && r.isAlias() && (_context == null || !_context.checkAlias(path, r)))
-                {
-                    if (LOG.isDebugEnabled())
-                        LOG.debug("resource={} alias={}", r, r.getAlias());
-                    return null;
-                }
+                if (LOG.isDebugEnabled())
+                    LOG.debug("Rejected alias resource={} alias={}", r, r.getAlias());
+                throw new IllegalStateException("Rejected alias reference: " + path);
             }
-            else if (_context != null)
-                r = _context.getResource(path);
+        }
+        else if (_context != null)
+        {
+            r = _context.getResource(path);
+            if (r != null)
+                return r;
+        }
 
-            if ((r == null || !r.exists()) && path.endsWith("/jetty-dir.css"))
-                r = getStylesheet();
+        if ((r == null || !r.exists()) && path.endsWith("/jetty-dir.css"))
+            r = getStylesheet();
 
-            return r;
-        }
-        catch (Exception e)
+        if (r == null)
         {
-            LOG.debug("Unable to get Resource for {}", path, e);
+            throw new FileNotFoundException("Resource: " + path);
```

Comment:
This is still the wrong exception.
The situation where r==nul does not represent file not found.  I not found file will return a resource that does not exist.
r==null (if it can happen???) represents something illegal or wrong!

Reply:
Your description of this PR says:
 > Refactored ResourceFactory to never return null and always throw an exception if unable to get/create/resolve the Resource
 > Resource.addPath(String) implementations can never return null now

So my reading of this is that r cannot be null!   So IllegalStateException would be more appropriate

Reply:
cancel that as Context.getResources can return null;